reactioncommerce/reaction
# fix: password-reset route not working correctly

Resolves #5726 
Impact: **critical**  
Type: **bugfix**

## Issue
The reset-password route does not correctly render the right component, making it impossible for you to reset your password unless you are under certain circumstances, like an admin who is logged in.

In addition, there was an issue related to this Password reset being a GQL mutation. On this page, there is no user, so `meteor-login-token` is not set. However, we were setting that value to `null`, instead of leaving it undefined. GQL was trying to use `null` as an auth check, which was throwing `Unauthorized` errors.

By leaving the `meteor-login-token` as undefined, we are able to use this mutation without being logged in / `null`.

## Solution
- Add a URL check to see if the route is `reset-password`, and show the form if it is.
- Add a check to make sure `meteor-login-token` is a value or `undefined`, but not `null`

## Notes
This does not fix the underlying problem, which is that the `reset-password` route doesn't register as it should and instead says `not-found` for the `name` of the route. It provides a temporary fix to the issue we are seeing until we can move authentication to it's own app.

## Breaking changes
None.


## Testing
1. Register for an account
1. Use the password reset email to reset your password
1. See that you land on the password reset page after clicking the link in the email
